src/Ydb.php

```diff
@@ -111,7 +116,10 @@ public function __construct($config = [], LoggerInterface $logger = null)
 
         if(isset($config['credentials'])){
             $this->iam_config['credentials'] = $config['credentials'];
-            $config['credentials']->setLogger($this->logger());
+            $this->iam_config['credentials']->setLogger($this->logger());
+            if (method_exists($this->iam_config['credentials'], "setAuthService")){
```

Comment:
Is it possible to create separate interface with method setAuthService, instead of direct check method by name?